### PR TITLE
Hard Audit: enable users to repay another account's borrows

### PR DESCRIPTION
**Pull request by denalimarsh**
Enables users to repay someone else's borrow position.

Tagging @downing034 as a reviewer since this impacts the REST API (additional required argument `owner` for MsgRepay).

### CODE REVIEW

File: x/hard/client/cli/tx.go [hunk 1]
```diff
@@ -119,12 +119,12 @@ func getCmdBorrow(cdc *codec.Codec) *cobra.Command {
 
 func getCmdRepay(cdc *codec.Codec) *cobra.Command {
 	return &cobra.Command{
-		Use:   "repay [1000000000ukava]",
+		Use:   "repay [amount] [owner]",
```
Comment (karzak) [inline x/hard/client/cli/tx.go:122]: Might be nicer to have owner be an optional flag, so the default syntax is 

```
kvcli tx hard repay 100000000ukava --from <key>
```

And to pay a different address' loan

```
kvcli tx hard repay 100000000ukava --owner kava1hgcfsuwc889wtdmt8pjy7qffua9dd2tralu64j --from <key>
```

In the default case, `from` becomes `owner`.

File: x/hard/client/rest/tx.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ func postRepayHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
 			return
 		}
 
-		msg := types.NewMsgRepay(req.From, req.Amount)
+		msg := types.NewMsgRepay(req.From, req.Owner, req.Amount)
```
Comment (karzak) [inline x/hard/client/rest/tx.go:100]: similar to CLI suggestion, could parse the request and if `Owner` is empty, use `From`.

**Review by karzak — CHANGES_REQUESTED** [review]
Minor client suggestions, otherwise LGTM

File: x/hard/client/cli/tx.go (hunk 1 above)
Comment (denalimarsh) [reply x/hard/client/cli/tx.go]: Refactored to optional `owner` flag in https://github.com/Kava-Labs/kava/pull/801/commits/a5b1ab86e6250b010c6abb256649b549e041fe6a

File: x/hard/client/rest/tx.go (hunk 2 above)
Comment (denalimarsh) [reply x/hard/client/rest/tx.go]: Updated REST in https://github.com/Kava-Labs/kava/pull/801/commits/4d595c50d47153e9a2a2a4be72475a32f63884d1

**Comment by denalimarsh** [comment]
Looks like master build is failing. This branch was passing CircleCI until master latest was merged in.

**Pull request merged**